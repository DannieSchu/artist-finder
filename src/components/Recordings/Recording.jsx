import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../global-styles/item.css';

const Recording = ({ title }) => (
  <section className={styles.item}>
    <h3>{title}</h3>
  </section>
);

Recording.propTypes = {
  title: PropTypes.string.isRequired
};

export default Recording;
