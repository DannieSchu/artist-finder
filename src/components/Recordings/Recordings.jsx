import PropTypes from 'prop-types';
import Recording from './Recording.jsx';
import withList from '../../utils/withList.jsx';

const Recordings = withList(Recording, 'recordings');

Recordings.propTypes = {
  recordings: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  })).isRequired
};

export default Recordings;
