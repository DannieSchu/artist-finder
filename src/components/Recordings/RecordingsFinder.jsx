import React from 'react';
import PropTypes from 'prop-types';
import Recording from './Recording.jsx';
import withList from '../../utils/withList.jsx';
import { useGetRecordings } from '../../hooks/getRecordings.jsx';

const RecordingsFinder = ({ match }) => {
  const { releaseId } = match.params;
  const { recordings, release, loading } = useGetRecordings(releaseId);
  const Recordings = withList(Recording, 'recordings');

  return (
    <Recordings 
      recordings={recordings} 
      heading={release} 
      loading={loading} 
    />
  );
};

RecordingsFinder.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      releaseId: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default RecordingsFinder;
