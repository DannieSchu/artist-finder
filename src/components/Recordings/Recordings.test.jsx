import React from 'react';
import { shallow } from 'enzyme';
import Recordings from './Recordings.jsx';

describe('Recordings', () => {
  it('matches a snapshot', () => {
    const recordings = [
      { title: 'some title',
        id: 'fneowigbw0' },
      { title: 'another title', 
        id: 'fneowigbw0' },
      { title: 'a third title',
        id: 'fneowigbw0' }
    ];

    const wrapper = shallow(
      <Recordings recordings={recordings} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
