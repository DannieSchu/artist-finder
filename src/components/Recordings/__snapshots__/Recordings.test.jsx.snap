// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recordings matches a snapshot 1`] = `
<section
  className="list"
>
  <h2>
    Results for '
    '
  </h2>
  <ul>
    <li
      key="fneowigbw0"
    >
      <Link
        to="/recording/fneowigbw0"
      >
        <Recording
          id="fneowigbw0"
          title="some title"
        />
      </Link>
    </li>
    <li
      key="fneowigbw0"
    >
      <Link
        to="/recording/fneowigbw0"
      >
        <Recording
          id="fneowigbw0"
          title="another title"
        />
      </Link>
    </li>
    <li
      key="fneowigbw0"
    >
      <Link
        to="/recording/fneowigbw0"
      >
        <Recording
          id="fneowigbw0"
          title="a third title"
        />
      </Link>
    </li>
  </ul>
</section>
`;
